src/geometry/PolySetBuilder.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <Reindexer.h>
+#include "Polygon2d.h"
+#include <memory>
+#include "boost-utils.h"
+class PolySet;
+
+class PolySetBuilder
+{
+  PolySet *ps;
```

Comment:
Not sure about this yet, but it could be cleaner to construct the polygon in the `result()` function (and perhaps rename it to `build()`, and retain the necessary data structures in this class.